File: GARI_Umbria_2026-08-03/GiasNG/src/app/Model/anagrafiche/addresses/Indirizzo.ts
import { CodiciNazioniISO3166 } from '../../metaschema/CodiciNazioniISO3166';
import { Istat } from '../../metaschema/Istat';
import {LatLng} from '../../GIS/Utility';

export class Indirizzo {
    codice: number;
    via: string;
    frazione: string;
    istatComune: Istat;
    cap: string;
    stato: CodiciNazioniISO3166;
    note: string;
    flag_cancellazione: boolean;
    geolocation: Required<LatLng> = {lat: 0, lng: 0, isValid: false};
}
export class IndirizzoDefault {
    CAP: string;
    Cod_Indirizzo: string;
    Comune: string;
    Indirizzo: string;
    IndirizzoCompleto: string;
    Localita: string;
    Provincia: string;
    Provincia_Sigla: string;
    Stato: string
}

File: GARI_Umbria_2026-08-03/DotNet/AgronicaCoreStd/AgronicaCoreModelsSTD/anagrafiche/Indirizzo.cs
using AgronicaCoreModelsSTD.Gis;
using AgronicaCoreModelsSTD.metaschema;
using System;
using System.Collections.Generic;
using System.Text;

namespace AgronicaCoreModelsSTD.anagrafiche
{
    public class Indirizzo
    {

        /// <summary>
        /// Cod_indirizzo su db
        /// </summary>
        public int codice { get; set; }

        public Istat istatComune { get; set; }

        public string via { get; set; }
        public string frazione { get; set; }
        public string cap { get; set; }
        public CodiciNazioniISO3166 stato { get; set; }
        public string note { get; set; }
        public LatLng geolocation { get; set; }
        public bool flag_cancellazione { get; set; }


        public Indirizzo()
        {
            flag_cancellazione = false;
            this.geolocation = new LatLng(0, 0);
        }

        public Indirizzo(int codice)
        {
            this.codice = codice;
            flag_cancellazione = false;
            this.geolocation = new LatLng(0, 0);
        }

        public Indirizzo(int codice, Istat istatComune, string via, string frazione, string cap, CodiciNazioniISO3166 stato, string note, bool flag_cancellazione) : this(codice)
        {
            this.istatComune = istatComune;
            this.via = via;
            this.frazione = frazione;
            this.cap = cap;
            this.stato = stato;
            this.note = note;
            this.flag_cancellazione = flag_cancellazione;
            this.geolocation = new LatLng(0, 0);
        }
        
        public Indirizzo(int codice, Istat istatComune, string via, string frazione, string cap, CodiciNazioniISO3166 stato, string note, bool flag_cancellazione, decimal lat, decimal lng) : this(codice)
        {
            this.istatComune = istatComune;
            this.via = via;
            this.frazione = frazione;
            this.cap = cap;
            this.stato = stato;
            this.note = note;
            this.flag_cancellazione = flag_cancellazione;
            this.geolocation = new LatLng(lat, lng);
        }
    }
}
